src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -394,9 +396,12 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		}
 		r.pandaCluster.SetDecommissionBrokerID(id)
 
-		if err = r.handleDecommissionInProgress(ctx, log); err != nil {
```

Comment:
Why `handleDecommissionInProgress` was changed to `handleDecommission`? The `podEviction` should not recommission?

Reply:
because the managed decommission workflow does not downscale the statefulset.

Reply:
Sure, but podEviction can be called in normal case when user don't want to run managed decommission. 

If user lower the replication factor following with reverting this back to previous number with this PR it will not handle it, am I missing something?

Reply:
Are you missing that this is only when we're doing a managed decommission?

Reply:
You are right. I missed that. When I started review the commit history wasn't clear. Resolved.